storage/store.go

```diff
@@ -65,17 +82,321 @@ type Store interface {
 	) error
 
 	// CollectionByID gets the collection (transaction IDs only) with the given ID.
-	CollectionByID(flowgo.Identifier) (flowgo.LightCollection, error)
+	CollectionByID(ctx context.Context, collectionID flowgo.Identifier) (flowgo.LightCollection, error)
 
 	// TransactionByID gets the transaction with the given ID.
-	TransactionByID(flowgo.Identifier) (flowgo.TransactionBody, error)
+	TransactionByID(ctx context.Context, transactionID flowgo.Identifier) (flowgo.TransactionBody, error)
 
 	// TransactionResultByID gets the transaction result with the given ID.
-	TransactionResultByID(flowgo.Identifier) (types.StorableTransactionResult, error)
+	TransactionResultByID(ctx context.Context, transactionID flowgo.Identifier) (types.StorableTransactionResult, error)
 
 	// LedgerViewByHeight returns a view into the ledger state at a given block.
-	LedgerViewByHeight(blockHeight uint64) *delta.View
+	LedgerViewByHeight(ctx context.Context, blockHeight uint64) *delta.View
 
 	// EventsByHeight returns the events in the block at the given height, optionally filtered by type.
-	EventsByHeight(blockHeight uint64, eventType string) ([]flowgo.Event, error)
+	EventsByHeight(ctx context.Context, blockHeight uint64, eventType string) ([]flowgo.Event, error)
+}
+
+type StorageBackendKeys interface {
+	StorageKey(key string) string
+	LatestBlockKey() []byte
+	BlockHeightKey(height uint64) []byte
+	IdentifierKey(id flowgo.Identifier) []byte
+	EventKey(blockHeight uint64, txIndex, eventIndex uint32, eventType flowgo.EventType) []byte
+}
+
+type StorageBackendData interface {
+	GetBytes(ctx context.Context, store string, key []byte) ([]byte, error)
+	SetBytes(ctx context.Context, store string, key []byte, value []byte) error
+	SetBytesWithVersion(ctx context.Context, store string, key []byte, value []byte, version uint64) error
+	GetBytesAtVersion(ctx context.Context, store string, key []byte, version uint64) ([]byte, error)
+}
+
+type StorageBackendKeysImpl struct {
```

Comment:
This `Impl` has very Java vibe. I would rather call it just `Keys` or something like that and then you also don't have to keep repeating keys in method names, so it can be `Keys.Storage()`. Any way you see fit but the `Impl` suffix doesn't feel idiomatic in Go. Also again I don't think this should be exported.